cmd/juju/application/config_test.go

```diff
@@ -287,37 +294,44 @@ func (s *configCommandSuite) TestResetConfigToDefault(c *gc.C) {
 func (s *configCommandSuite) TestBlockSetConfig(c *gc.C) {
 	// Block operation
 	s.fake.err = common.OperationBlockedError("TestBlockSetConfig")
-	ctx := cmdtesting.ContextForDir(c, s.dir)
-	code := cmd.Main(application.NewConfigCommandForTest(s.fake), ctx, []string{
+	cmd := application.NewConfigCommandForTest(s.fake)
+	cmd.SetClientStore(application.NewMockStore())
+	_, err := cmdtesting.RunCommandInDir(c, cmd, []string{
 		"dummy-application",
 		"--file",
-		"testconfig.yaml"})
-	c.Check(code, gc.Equals, 1)
-	// msg is logged
-	stripped := strings.Replace(c.GetTestLog(), "\n", "", -1)
-	c.Check(stripped, gc.Matches, ".*TestBlockSetConfig.*")
+		"testconfig.yaml",
+	}, s.dir)
+	c.Assert(err, gc.ErrorMatches, `(.|\n)*All operations that change model have been disabled(.|\n)*`)
+	c.Check(c.GetTestLog(), gc.Matches, "(.|\n)*TestBlockSetConfig(.|\n)*")
```

Comment:
I can never remember the meaning of all those flags! I think the explicit form is probably better because it's more obvious what's going on.